pkgs/by-name/wi/winbox4/package.nix

```diff
@@ -0,0 +1,110 @@
+{
+  autoPatchelfHook,
+  copyDesktopItems,
+  fetchurl,
+  fontconfig,
+  freetype,
+  lib,
+  libGL,
+  libxkbcommon,
+  makeDesktopItem,
+  makeWrapper,
+  stdenvNoCC,
+  unzip,
+  writeShellApplication,
+  xorg,
+  zlib,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "winbox";
+  version = "4.0beta6";
+
+  src = fetchurl {
+    name = "WinBox_Linux-${finalAttrs.version}.zip";
+    url = "https://download.mikrotik.com/routeros/winbox/${finalAttrs.version}/WinBox_Linux.zip";
+    hash = "sha256-PCb1jQDzSblYdf+avK/IYQeYgD19Jx8/Kwdg9/EgxNs=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    # makeBinaryWrapper does not support --run
+    makeWrapper
+    unzip
+  ];
+
+  buildInputs = [
+    fontconfig
+    freetype
+    libGL
+    libxkbcommon
+    xorg.libxcb
+    xorg.xcbutilimage
+    xorg.xcbutilkeysyms
+    xorg.xcbutilrenderutil
+    xorg.xcbutilwm
+    zlib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 "assets/img/winbox.png" "$out/share/pixmaps/winbox.png"
+    install -Dm755 "WinBox" "$out/bin/WinBox"
+
+    wrapProgram "$out/bin/WinBox" --run "${lib.getExe finalAttrs.migrationScript}"
+
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "winbox";
+      desktopName = "Winbox";
+      comment = "GUI administration for Mikrotik RouterOS";
+      exec = "WinBox";
+      icon = "winbox";
+      categories = [ "Utility" ];
+    })
+  ];
+
+  migrationScript = writeShellApplication {
+    name = "winbox-migrate";
+    text = ''
+      XDG_DATA_HOME=''${XDG_DATA_HOME:-$HOME/.local/share}
+      sourceFile="$HOME/.wine/drive_c/users/$USER/Application Data/Mikrotik/Winbox/Addresses.cdb"
```

Comment:
Hmm, does this path actually work for you? In winbox3 we override home:
https://github.com/NixOS/nixpkgs/blob/f200f28aef5e55ccf44a8496d1b7496e877a0ea0/pkgs/tools/admin/winbox/default.nix#L27

For me the correct path would be:
`$HOME/.local/share/winbox/wine/drive_c/users/$USER/AppData/Roaming/Mikrotik/Winbox/Addresses.cdb`

Reply:
And - no clue why the difference between AppData vs Application Data subpaths. Might have to do with version of Windows that was originally "faked", but I have no `Application Data` directory in there (which is meant to be an alias for `AppData/Roaming` on new Windows versions as I understand it.

Reply:
> Hmm, does this path actually work for you? In winbox3 we override home

I had written the migration script based on the original forum post and my local (manual!) installation of WinBox 3. I'll adjust this to cover the default path used by our winbox3 package.

Reply:
> https://github.com/NixOS/nixpkgs/blob/f200f28aef5e55ccf44a8496d1b7496e877a0ea0/pkgs/tools/admin/winbox/default.nix#L27

This is an odd revision. The derivation does not do this anymore in master/unstable. WinBox 3 gets installed into your default wine prefix

Reply:
I have refactored the migration script to support both paths. Please re-review. In my case, I was able to test this easily by running the migration script directly and deleting the target directory before running it:

```console
$ rm -rv ~/.local/share/MikroTik/
$ nix run .#winbox4.migrationScript
```

Reply:
Yup, looks to be working for me now